lib/origen_testers/smartest_based_tester/base/test_methods/base_tml.rb

```diff
@@ -12,6 +12,11 @@ def initialize(test_methods)
           end
 
           def method_missing(method, *args, &block)
+            tml_class_name = self.class.name.to_s.split('::')[-1]
+            # This enables passing camel case test methods for dc_tml and ac_tml
+            unless tml_class_name == 'CustomTml'
+              method = method.to_s.underscore.to_sym if camelcase?(method)
```

Comment:
I don't think this check for camelcase should be here.
I think this would be better written as something along the lines of - "if the test method as given does match a definition, the lowercase and underscore it and try again".
One of the concerns I have is corner cases where I have seen test method names be inconsistent (I think even in ADV code), between using camelCase and PascalCase.
Taking a less camel case specific approach per above should handle that.

Reply:
@ginty OK let me rework it.

Reply:
@ginty Is this what you envisioned?

~~~ruby
          def method_missing(method, *args, &block)
            if definitions[method]
              instantiate_test_method(method, args)
            else
              method = method.to_s.underscore.to_sym
              if definitions[method]
                instantiate_test_method(method, args)
              else
                super
              end
            end
          end
~~~

Reply:
Yep, exactly like that